src/Symfony/Component/Routing/RouteContext.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Routing;
+
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ */
+class RouteContext
+{
+    private $name;
+    private $parameters;
+    private $referenceType;
+
+    /**
+     * @param string $name          The name of the route
+     * @param mixed  $parameters    An array of parameters
+     * @param int    $referenceType The type of reference to be generated (one of the constants)
+     */
+    public function __construct($name, $parameters = array(), $referenceType = UrlGeneratorInterface::ABSOLUTE_PATH)
```

Comment:
I copied exactly the signature from `UrlGeneratorInterface::generate()`. I think it should match that, no?